src/common/plugins.js

```diff
@@ -0,0 +1,118 @@
+'use strict'
+
+const { app, nativeImage } = require('electron')
+const { join } = require('path')
+const { warn, verbose, logger } = require('./log')
+const { uniq, pluck } = require('./util')
+const { promises: jsonld } = require('jsonld')
+
+class Plugins {
+  constructor(root, plugins = []) {
+    this.root = root
+    this.plugins = plugins
+    this.instances = []
+    this.loadPaths = [
+      join(this.root, 'node_modules'),
+      this.root
+    ]
+    this.isRenderer = process.type === 'renderer'
+  }
+
+  get context() {
+    return {
+      logger,
+      jsonld,
+      nativeImage
+    }
+  }
+
+  contract(Plugin, config) {
+    return new Plugin(config, this.context)
+  }
+
+  get packages() {
+    return uniq(this.plugins.map(p => p.package))
+  }
+
+  // try to load the package from one of the paths
+  loadPackage(name) {
+    let pkg, hooks
+    for (let i = 0; i < this.loadPaths.length; i++) {
+      const path = join(this.loadPaths[i], name)
+      try {
+        pkg = require(path)
+        hooks = require(join(path, 'package.json')).hooks || {}
+      } catch (error) {
+        continue
```

Comment:
Let's check the `error.code` here. If it is a module load error, the plugin is missing; otherwise it will be a faulty plugin. In both cases we want to log the error.

Reply:
Well, not the load error, because would be logged below; but in case of a faulty plugin, we want to log the error.